### ISSUE TITLE
Sorting by custom store view scope attribute

### ISSUE THREAD

**Issue by aglumbik** [Smile-SA/elasticsuite#838]
Hello,
we have problem with sorting our categories by custom product attribute for which we setting values on store view level. For example we have attribute "Points" which we calculate everyday by real sales. And for each store view, each product get different value. We have dropdown attribute with options: 1,2,3,4,5..5000 and according to result we assign option on store view level. This works in clean Magento installation (I tested it), but with Smile modules, it works only if I set attribute on Default level.

### Preconditions

Magento Version : 2.2.2

ElasticSuite Version : 2.5.6

Environment : Developer

Elasticsearch: 5.4

Third party modules : All disabled except Magento core and Smile

### Steps to reproduce

1. Install clean Magento 2.2.2
2. Install Smile over composer
3. Create sortable dropdown attribute with store view scope
4. Assign values for several products on Store view level
5. Try do the same thing on Default level

### Expected result

1. I expecting that it will work like in clean Magento installation. Products which has assign value of sort attribute get on top, and the rest behind.

### Actual result

1. Assignation of sort attribute on store view level has no effect. Only assignation on Default level works.

**Comment by romainruaud** [comment]
Hello @aglumbik 

I'm not sure to understand everything properly.

Why having a dropdown with options if you attribute is basically a "number" which is calculated everyday ? Did you try to use a decimal attribute instead ?

**Comment by afoucret** [comment]
Using a select attribute is a poor design option since they are considered as string from the search engine point of view. and sorted in the lexicographic order (1,10,100,11,2,210,...).

You should better use a integer attribute.


BR,

**Comment by simonrl** [comment]
Maybe helpful:
I had the same problem with an attribute that is configured as a textfield input with numerical validation (so no dropdown).

I noticed that the sorting does not work if there's no value at all in the global scope (I added the values using an import script, so this is a special use case).
As soon as I added the value in the global scope (for example by simply saving a product, or by adding it in the database) and re-indexed, the store view values got recognized correctly and sorting was ok.
